packages/kbn-esql-ast/src/ast_walker.ts

```diff
@@ -346,14 +351,18 @@ function getConstant(ctx: ConstantContext): ESQLAstItem {
     ctx instanceof StringArrayLiteralContext
   ) {
     const values: ESQLLiteral[] = [];
+
+    // @todo: double check here
     for (const numericValue of ctx.getTypedRuleContexts(NumericValueContext)) {
+      const isDecimal = numericValue.decimalValue() !== undefined;
```

Comment:
Just wanted to double check that `const isDecimal = numericValue.decimalValue() !== undefined` is a valid way to check. Here the assumption is that if it's numeric but not decimal, it has to be an integer.

Reply:
You know as much as me on this. It seems to work!